### DISCUSSION TITLE
ETA for unnecessary select optimization

### DISCUSSION THREAD

**Discussion by michael-land** [prisma/prisma#5632] [Q&A]
Hi, prisma team, is there ETA for this bug? 

https://github.com/prisma/prisma/issues/4246

This unnecessary select make our app super slow. Our use case is really simple,  the script have to upsert 10000 records every 10 minutes use `prisma.model.upsert`.

For each upsert, prisma will take about 500ms ~ 1s to execute, (compare to node-postgres which only take 50ms ~ 100ms). This performance issue make prisma is not suitable for most of our script. 

I wonder is there any plan for this optimization? If it is not on the roadmap, I might need to migrate most scripts to raw sql.

```ts
await this.prisma.product.upsert({
  select: { id: true },
  where: { id: product.id },
  update: { id: product.id },
  create: { id: product.id },
});

prisma:query BEGIN
prisma:query SELECT "public"."product"."id" FROM "public"."product" WHERE "public"."product"."id" = $1 OFFSET $2
prisma:query SELECT "public"."product"."id" FROM "public"."product" WHERE "public"."product"."id" = $1
prisma:query UPDATE "public"."product" SET "gtin" = $1, "unit_price" = $2, ...and more WHERE "public"."product"."id" IN ($21)
prisma:query SELECT "public"."product"."id" FROM "public"."product" WHERE "public"."product"."id" = $1 LIMIT $2 OFFSET $3
prisma:query COMMIT
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
